Add mark as read button

benonymity opened the pull request:
Simple addition of button to mark book as read, fixes #61. Though with the download button that area might start to get cramped on smaller devices.

advplyr commented:
There is additional steps to take here for downloaded items.

There are several cases:
1. Local library item not connected to a server item: Call native update media progress only
2. Local library item connected to a server item: Call native update to media progress, IF currently connected to the server of this library item then also make the server request.
3. Library item on server: Make server request

benonymity commented:
I unfortunately don't have access to an Android device to test the local progress updates for the read button--does it look like it works? It does in theory but there might be some issue I can't see in practice. I can confirm it does definitely sync to the server at the very least and appears properly in the iOS app. If so should be ready to merge!

advplyr commented:
Did you run this in the browser using npm run dev? Or were you looking at it on iphone?

benonymity commented:
No errors with `npm run dev`, and also works when built on iOS. I just downloaded Android Studio, but I can't emulate a device as my CPU doesn't support virtualization, so can't confirm it works there though I imagine so.

advplyr commented:
Does it look right for you? I did an npm run dev and the buttons were different sizes and overflowing depending on the device.

benonymity commented:
Seems I tested it on the one device where the natural sizing worked haha, just added proper classes for CSS. 
![Read_Btn](https://user-images.githubusercontent.com/62854267/165602251-1f051b67-016c-4a34-93b2-4863a340a1bf.jpeg) Should display properly now.

advplyr commented on pages/item/_id.vue:
```diff
@@ -351,6 +354,52 @@ export default {
+    },
+    async toggleFinished() {
+      this.isProcessingReadUpdate = true
+      if (this.isLocal || this.localEpisode) {
```
This button should not show for podcasts.

advplyr commented:
Something about those buttons not being the same height is throwing me off. Maybe we make the play button skinnier or meet in the middle somewhere.

benonymity commented:
Removed podcast reference and made CSS consistent with server UI for the play button, so now it's the same size at the read button. Does this look okay?

![shorter_playbtn](https://user-images.githubusercontent.com/62854267/165810751-2a537e8a-1bcb-422d-9fd2-000e1b4d8679.jpeg)

advplyr commented on pages/item/_id.vue:
```diff
@@ -26,17 +26,18 @@
             <span class="material-icons">auto_stories</span>
             <span v-if="!showPlay" class="px-2 text-base">Read {{ ebookFormat }}</span>
           </ui-btn>
+          <ui-read-icon-btn :disabled="isProcessingReadUpdate" :is-read="userIsFinished" class="mx-0.5 flex items-center justify-center" @click="toggleFinished" />
```
We can't show this for podcasts

benonymity replied:
I didn't think this page was accessible for a podcast item, they have their own page with independent buttons to mark as read correct? But if this does ever get used for a podcast I added `v-if="!isPodcast"`. Was that what you meant?

advplyr commented:
![image](https://user-images.githubusercontent.com/67830747/165843291-f82651af-a984-4dfb-ac0e-8ac9913f20a1.png)

benonymity commented:
Okay, this *should* work... I resized the read button slightly to fit alongside the old play button. The resizing doesn't impact the podcast view. I also disabled the download button for podcasts on iOS because that just threw errors. Sorry for all the mistakes and back and forth!

advplyr commented:
No problem! Are you running this in your browser to test? I'm getting the same layout issues.

![image](https://user-images.githubusercontent.com/67830747/165858315-91a57a21-886e-4e24-adf0-2540b988bd2f.png)

benonymity commented:
Woah! That's so much easier, I was unaware of that dev tool to change display size! I've been compiling the app in XCode and sending it to my iPhone each time I made a change to test...hot reloading is incredible! Thanks for showing me the light lol. *Now* it should actually be working, I fixed the padding and tested it at a bunch of different resolutions and it seems to work well. Also fixed the download icon being out of line with the read button icon in the podcast table view.

advplyr commented:
Is this intentional being on 2 lines? I think we can shorten this to just "Stream" and "Play". The difference in wording there helps to distinguish between when you are playing audio from your server verses playing a locally downloaded audio file.

![image](https://user-images.githubusercontent.com/67830747/166118122-e28af19b-4a4d-4619-b37d-dbc26f652b65.png)

benonymity commented:
Good idea! Done.

advplyr commented:
Awesome, great job. I tested on android and it works for local downloads too.
